Loop/Models/RetrospectiveCorrection/IntegralRetrospectiveCorrection.swift

```diff
@@ -179,7 +179,10 @@ class IntegralRetrospectiveCorrection: RetrospectiveCorrection {
             
             // Overall glucose effect calculated as a sum of propotional, integral and differential effects
             proportionalCorrection = IntegralRetrospectiveCorrection.proportionalGain * currentDiscrepancyValue
-            differentialCorrection = IntegralRetrospectiveCorrection.differentialGain * differentialDiscrepancy
+            // Differential effect added only when negative, to avoid upward stacking with momentum, while still mitigating slugeshness of retrospective correction when discrepancies start decreasing
```

Comment:
With this change, IRC is still accumulating increasing positive effects as time goes on, right?  It's just eliminating this "gain" on the positive side, right?

Also, small typo: "slugeshness" -> sluggishness.

Reply:
Yes, IRC is still accumulating increasing positive effects as time goes on; this is just eliminating an additional positive effect when the most recent discrepancy is higher than the one before. This is likely to occur when glucose is increasing quickly (while there are no carbs on board, so RC effects are present), which is why stacking with momentum can happen. In the opposite direction, this effect is safe and beneficial - reduces the retrospective effect more quickly when discrepancies are decreasing.

Reply:
@dm61 I'm interested in your thoughts on leaving in the differential term for negative differentialDiscrepancy and what you want it to do, both to understand it and to help evaluate its effectiveness.

Reply:
@bedtime4bonzos consider a scenario when discrepancies remain positive for extended periods of time, for example, due to temporarily reduced insulin sensitivity. or some other unmodeled factor pushing glucose upwards. IRC accumulates such effects, which results in a larger (positive) retrospective correction. This is all fine as long as the unmodeled factor persists. However, once the unmodeled factor goes away, for example, insulin sensitivity goes back up, IRC tends to be sluggish as it takes some time for the accumulated discrepancies to decay. That's where the differential term comes into play - reduces the retrospective correction in response to discrepancies that are _decreasing_.  As a result, IRC with the differential term (in the negative direction) is less sluggish (and thereby safer) when the unmodeled effects go away. In the positive direction, the differential term, especially in combination with the. momentum, can result in too large positive corrections, which is why this has been removed in this IRC update.